### DISCUSSION TITLE
What is the purpose of the "clk" parameter in the Init function of the algorithm?

### DISCUSSION THREAD

**Discussion by MarianSavchuk** [ARM-software/CMSIS_5#1643] [Q&A] [answered]
Hello Team,

I am seeking clarification regarding the clk parameter in the [Init ](https://github.com/ARM-software/CMSIS_5/blob/develop/Device/_Template_Flash/FlashPrg.c#L61)function.
I am unsure if this parameter is used to establish the core frequency or if it is used to adjust the timing calculations for flash programming based on the device's clock speed. 
Can you please explain its actual purpose?

Best regards,
Marian

**Comment by ghost** [comment, marked as answer]
Hello @MarianSavchuk , the intention of the **clk** parameter was to provide the current core frequency to the flash algorithms in case adjustments are necessary.
To be honest up to now I never used this parameter.

**Comment by MarianSavchuk** [comment]
Hello @GuentherMartin,
Thanks a lot for the explanation!
Best regards
